app/partials/app.cjsx

```diff
@@ -19,9 +19,12 @@ PanoptesApp = React.createClass
     geordi: @geordiLogger
 
   getEnv: ->
+    shell_env = process.env.NODE_ENV
     reg = /\W?env=(\w+)/
     browser_env = window?.location?.search?.match(reg)
-    @state?.env || browser_env?[1] || 'staging'
+    env = shell_env || browser_env?[1] || 'staging'
+    @setState env: env
```

Comment:
funnily enough, what you describe is what I had originally, but then I was worried it wasn't robust enough. I guess since shell_env cannot change after launch, and as you say getInitialState will call this method anyway, then we will be fine. have updated.